src/com/palmergames/bukkit/towny/object/Nation.java

```diff
@@ -712,14 +715,47 @@ public Resident getKing() {
 		return capital.getMayor();
 	}
 
-	public void addMetaData(CustomDataField md) {
-		super.addMetaData(md);
+	@Override
+	public MetaMap getMetadata() {
+		return metadata;
+	}
+
+	@Override
+	public boolean hasMeta() {
+		return metadata != null;
+	}
+
+	@Override
+	public void setMetadata(String str) {
+		if (metadata == null)
+			metadata = new MetaMap();
+
+		String[] objects = str.split(";");
+		for (String object : objects) {
+			CustomDataField<Object> field = CustomDataField.load(object);
+			metadata.put(field.getKey(), field);
+		}
+	}
```

Comment:
Looks like this might also need a `TownyUniverse.getInstance().getDataSource().saveNation(this);` added to it, no?

If so, the other objects are also missing save calls in their setMetaData methods.

Reply:
Yeah I took your suggestion, originally the object would save its meta within the TownyAdminCommand.java, but I realized that developers need the changes to be saved as well.